docs/src/howToGetStarted.md

```diff
@@ -1,25 +1,268 @@
 
-10 cell surface markers (lineage markers) are used to identify different cell populations:
-    - `PBMC8_panel.xlsx` (with Antigen name and columns for lineage markers and functional markers)
-    - `PBMC8_metadata.xlsx` (file names, sample id, condition and patient id)
+You can always collect the scattered data back into a matrix (if it fits to
+your RAM) with [`distributed_collect`](@ref), and utilize many other functions
+to manipulate it, including e.g. [`distributed_mapreduce`](@ref) for easily
+running parallel computations, or [`distributed_export`](@ref) for saving and
+restoring the dataset paralelly.
 
-Before running this minimum working example, make sure to use the package:
+## Minimal working example
+
+First, load GigaSOM:
 
 ```julia
 using GigaSOM
 ```
 
-## Input and output
+We will create a bit of randomly generated data for this purpose. This code
+generates a 4D hypercube of size 10 with gaussian clusters at vertices:
+
+```julia
+d = randn(10000,4) .+ rand(0:1, 10000, 4).*10
+```
+
+The SOM (of size 20x20) is created and trained as such:
+
+```julia
+som = initGigaSOM(d, 20, 20)   #this requires random numbers, use Random.seed!() for reproducibility
+som = trainGigaSOM(som, d)
+```
+
+You can now see the SOM codebook (your numbers will vary):
+
+```julia
+som.codes
+```
+
+```
+400×4 Array{Float64,2}:
+ -0.361681    -0.57191     0.140438   9.99224
+ -1.111        1.60277    -0.209706   9.96805
+ -1.23305      7.58148    -0.445886   9.76316
+ -0.285692     9.80184    -1.12107    9.85507
+ -0.197007    10.8793     -0.649294   9.89448
+ -0.334737    11.0858      0.213889   9.93479
+  0.00282155  11.0725      0.718114  10.2714
+ -0.333398    10.1315      1.14412   10.564
+ -0.0124202    1.48128     8.35741   10.72
+ -0.0084074    0.0150858   9.91007   11.5361
+  ⋮
+```
+
+This information can be used to categorize the dataset into clusters:
+
+```julia
+mapToGigaSOM(som, d)
+```
+
+In the result, `index` is a cluster ID for the original datapoint from `d` at
+the same row.
+```
+10000×1 DataFrames.DataFrame
+│ Row   │ index │
+│       │ Int64 │
+├───────┼───────┤
+│ 1     │ 381   │
+│ 2     │ 178   │
+│ 3     │ 348   │
+│ 4     │ 379   │
+│ 5     │ 80    │
+│ 6     │ 146   │
+│ 7     │ 57    │
+⋮
+```
+
+(With a different SOM initialization and dataset, the cluster assignment and
+results will be different. For clustering to a smaller number of clusters, you
+can use `som.codes` for metaclustering, as in FlowSOM.)
+
+Finally, you can create a 2D picture of the result using EmbedSOM:
+```julia
+e = embedGigaSOM(som,d)
+```
+
+```
+10000×2 Array{Float64,2}:
+  1.41575  18.5282
+ 17.4483    7.4137
+  6.88243  17.5722
+ 17.654    18.0348
+ 17.594     3.15645
+  5.27181   8.61096
+ 15.9708    2.8124
+  6.19637   9.05302
+  1.49358   7.19198
+ 16.596     7.75608
+  ⋮
+```
+
+This can be plotted out using any of the plotting libraries:
+```julia
+using Gadfly
```

Comment:
Maybe we shall provide instructions or a link to the installation instructions of these dependencies.